db/range_index.go

```diff
@@ -54,29 +54,38 @@ type (
 
 	// rangeIndex is RangeIndex implementation based on bolt DB
 	rangeIndex struct {
-		db         *bolt.DB
-		numRetries uint8
-		bucket     []byte
+		*boltDB
+		bucket []byte
 	}
 )
 
 // NewRangeIndex creates a new instance of rangeIndex
-func NewRangeIndex(db *bolt.DB, retry uint8, name []byte) (RangeIndex, error) {
-	if db == nil {
-		return nil, errors.Wrap(ErrInvalid, "db object is nil")
+func NewRangeIndex(kv KVStore, name, init []byte) (RangeIndex, error) {
+	if kv == nil {
+		return nil, errors.Wrap(ErrInvalid, "KVStore object is nil")
+	}
+	db, ok := kv.(*boltDB)
+	if !ok {
+		return nil, errors.Wrap(ErrInvalid, "range index can only be created from boltDB")
```

Comment:
if we throw error, we shouldn't let kv to be type of KVStore.

Reply:
what do you think is better way/method?
we need to create range index from dao.kvstore, which is of type KVStore